miso/App.hs

```diff
@@ -0,0 +1,48 @@
+module App (start) where
+
+import Control.Monad (void)
+import Control.Monad.Trans.Maybe (MaybeT (MaybeT, runMaybeT))
+import JSDOM (currentDocument)
+import JSDOM.Generated.Node (removeChild)
+import JSDOM.Generated.NonElementParentNode (getElementById)
+import JSDOM.Generated.ParentNode (getFirstElementChild)
+import Miso hiding (getElementById)
+import Miso.String (MisoString)
+
+start :: JSM ()
+start = do
+    maybe (consoleLog "placeholder node not found") (void . uncurry removeChild) =<< runMaybeT do
+        doc <- MaybeT currentDocument
+        parent <- MaybeT $ getElementById doc ("miso" :: MisoString)
+        child <- MaybeT $ getFirstElementChild parent
+        pure (parent, child)
```

Comment:
This replicates the functionality of the Elm example: a "not working" placeholder text is initially present, and removed when the Elm/Miso runtime has finished setting up. Things are complicated by the fact that, unlike Elm, Miso inserts itself as a child of the given node, rather than taking it over. So we have to manually call some DOM functions to remove the placeholder.

I've made this a separate commit seeing as the increased complexity in terms of dependencies and imports is such that it might honestly not be worth it, so we may wish to revert.

Alternatively, we could look in to the possibility of adapting Miso to have the option of replicating the Elm behaviour.

Reply:
The need for a `:: MisoString` annotation is a common problem when using `jsaddle-dom`, which tends to accept arbitrary string-like types, and thus doesn't play well with `OverloadedStrings` in its current form.

GHC 9.12's `NamedDefaults` would help here.